File: Binary Tree to BST[08-September-2023]/Solution.java
//   >> JAVA CODE <<

class Solution
{
    void inorder(Node root,ArrayList<Integer> list)
    {
        if(root==null) return;
        
        inorder(root.left,list);
        
        list.add(root.data);
        
        inorder(root.right,list);
    }
    void convert(Node root,ArrayList<Integer> list,int[] index)
    {
        if(root==null) return;
        
        convert(root.left,list,index);
        
        root.data=list.get(index[0]);
        index[0]++;
        
        convert(root.right,list,index);
    }
    Node binaryTreeToBST(Node root)
    {
       ArrayList<Integer> inorderList=new ArrayList<>();
       
       inorder(root,inorderList);
       
       Collections.sort(inorderList);
       
       convert(root,inorderList,new int[]{0});
       
       return root;
    }
}

File: Binary Tree to BST[08-September-2023]/Solution.cpp
//  >> CPP CODE <<

class Solution{
  public:
    void inorder(Node* root, vector<int>& list) {
        if (root == nullptr) return;
        
        inorder(root->left, list);
        
        list.push_back(root->data);
        
        inorder(root->right, list);
    }

    void convert(Node* root, vector<int>& list, int& index) {
        if (root == nullptr) return;
        
        convert(root->left, list, index);
        
        root->data = list[index];
        index++;
        
        convert(root->right, list, index);
    }

    Node* binaryTreeToBST(Node* root) {
        vector<int> inorderList;
       
        inorder(root, inorderList);
       
        sort(inorderList.begin(), inorderList.end());
       
        int index = 0;
        convert(root, inorderList, index);
       
        return root;
    }
};